### DISCUSSION TITLE
Id column in database table

### DISCUSSION THREAD

**Discussion by excellencemichel** [serverpod/serverpod#351] [Q&A] [answered]
Let user a possibility to create or generate Id for database table.
In some case we want to use other types like UUID sytem for id.

**Comment by yahu1031** [comment]
Yes, even I'm trying to do it.

**Comment by excellencemichel** [comment]
Okay, nice!
Thank you !

**Comment by vlidholt** [comment, marked as answer]
Best practice for this scenario is to create a second column and set a unique index for it.

**Reply by excellencemichel** [reply]
> Best practice for this scenario is to create a second column and set a unique index for it.

Okay Cool.
Thank you

**Reply by oco-adam** [reply]
Hi @vlidholt, I'm just checking that this best practice still applies and that it's still not possible to specify the type of the id field to be a String, for example?

**Reply by vlidholt** [reply]
The id field will always be an int. You should still create a second column with a unique index.

**Reply by oco-adam** [reply]
Thanks for the confirmation

**Comment by perumalrollfi** [comment]
Hi @vlidholt,
We tried to use Uuid as the variable type in yaml, The generated class throws error .  Can you please check and advise?
![image (3)](https://user-images.githubusercontent.com/93102280/217700059-949a67a9-21f9-44ac-a091-eed4c50d96f2.png)

Thanks

**Reply by yahu1031** [reply]
I don't think, We can use 3rd party package datatypes in the YAML, We generate the datatypes either from dart or serverpod. So they won't be recognized/imported.

**Reply by vlidholt** [reply]
It would be nice to add support for Uuid (though you can still store it as a String for now).

I added an issue here: https://github.com/serverpod/serverpod/issues/635

**Reply by yahu1031** [reply]
If we can add support to uuid then we can support all type of datatypes accordingly(generic)

**Reply by davideroffo** [reply]
That issue has been closed as complete. When do you plan to create a new release containing that new feature? I'm really interested in having it during the development of my app.

**Reply by vlidholt** [reply]
I think the next release will be in a week or so!

**Reply by muhammedrashidm** [reply]
Is this updated, I can't find anything on documentations
